### PR TITLE
Build stand alone executable

### PR DESCRIPTION
**Pull request by dagheyman**
Use PyInstaller to build ykman as a single executable on macOS and Windows.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,39 +1,74 @@
...
-  - "3.6"
-
-cache:
-  directories:
-    - $HOME/.cache/pip
+matrix:
+    include:
+        - os: linux
+          sudo: required
+          python: 3.6
```
Comment (emlun) [inline .travis.yml]: We should probably keep building with Python 2 while we support it, no?

File: .travis.yml [hunk 2]
```diff
@@ -1,39 +1,74 @@
...
+    fi
 
 install:
-  - pip install pre-commit
-  - pip install -e .
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "linux" ]]
+    then
+      pip install pre-commit
+      pre-commit run --all-files
```
Comment (emlun) [inline .travis.yml]: Should `pre-commit run` not be in `script:` instead?

File: .travis.yml [hunk 3]
```diff
@@ -1,39 +1,74 @@
...
+      pip install -e .
+      python setup.py test
+    fi
 
 script:
-  - pre-commit run --all-files
-  - python setup.py test
```
Comment (emlun) [inline .travis.yml]: Looks like the `python setup.py test` step was lost.

File: appveyor.yml [hunk 4]
```diff
@@ -5,39 +5,59 @@
...
 environment:
-  LIBU2F_HOST_VERSION: "1.1.4"
   YKPERS_VERSION: "1.19.0"
   LIBUSB_VERSION: "1.0.21"
+  YKPERS_BASEURL: https://developers.yubico.com/yubikey-personalization/Releases
 
   matrix:
-    - PYTHON: "C:\\Python27"
-    - PYTHON: "C:\\Python35"
-    - PYTHON: "C:\\Python36"
```
Comment (emlun) [inline appveyor.yml]: Again, I think we'd getter keep CI testing Python 2 for now...

File: appveyor.yml [hunk 5]
```diff
@@ -5,39 +5,59 @@
...
+  - ps: Copy-Item ".\libusb\$env:LIBUSB_DLL_FOLDER\dll\*.dll" "." -Force -verbose
 
 build_script:
-  - pip install -e .
-  - ykman -v
-
-test_script:
-  - python setup.py test
+  - pip install .
+  - pip install pyinstaller
```
Comment (emlun) [inline appveyor.yml]: Could `pip install pyinstaller` be moved to `install:`?

File: appveyor.yml [hunk 6]
```diff
@@ -5,39 +5,59 @@
...
   - 7z x libusb-%LIBUSB_VERSION%.7z -o".\libusb"
-  - set PATH=%PATH%;%APPVEYOR_BUILD_FOLDER%\libusb\MS32\dll
+  - ps: Copy-Item ".\libusb\$env:LIBUSB_DLL_FOLDER\dll\*.dll" "." -Force -verbose
 
 build_script:
-  - pip install -e .
-  - ykman -v
-
-test_script:
-  - python setup.py test
```
Comment (emlun) [inline appveyor.yml]: Why was the test step removed?

**Comment by dagheyman** [comment]
I tried to address most of the comments. Both appveyor and travis now has conditional deploys, so we can test as many python versions as we want but only build artifacts on some of them.

**Pull request merged**